### ISSUE TITLE
Ring Fit Adventure - Freeze when entering the "My Menu" screen

### ISSUE THREAD

**Issue by stealsfromhobos** [yuzu-emu/yuzu#10761]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

Mainline 1468/EA 3669

### Description of Issue

When playing Ring Fit Adventure, if I try to open the ingame options menu by pressing B the game hangs.  It plays the menu transition but the display gets stuck afterwards with the music still playing and FPS counter saying 0.

This happens with both Mainline and Early Access.  OpenGL and Vulkan.

### Expected Behavior

The "My Menu" screen should come up without issue when pressing B with an adventure profile selected.

### Reproduction Steps

Once in the Ring Fit main menu, click on an adventure profile and press B.  The game will hang when trying to open the next menu.

### Log File

[yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/11737976/yuzu_log.txt)

### System Configuration

CPU:Intel Core i7-8700k
GPU/Driver: GeForce RTX 3070 Driver ver. 531.68
RAM: 16GB
OS: Windows 10

**Comment by EgoBizarro** [comment]
Yeah, I can confirm the issue. Maybe it used the web applet or something?

**Comment by testRob-hash** [comment]
Yes, I have this problem too, does it require an internet connection to open?

**Comment by stealsfromhobos** [comment]
> Yes, I have this problem too, does it require an internet connection to open?
There might be a leaderboard in there that requires internet (See pic)?  But there's other important things like difficulty settings that I don't think would be gated behind an internet connection.  When I played on Switch I kept it offline and the menu worked fine from what I can recall.  My Switch is broken now so I can't test it.
![image](https://github.com/yuzu-emu/yuzu/assets/46424226/cf843e76-2371-4b3d-9e97-ea62b4c57458)

**Comment by EgoBizarro** [comment]
Didn't the leaderboard stuff only come with that big update?
Because I know it also didn't work on yuzu when I had forgotten to install the update.

**Comment by DaosArg** [comment]
Hello. Yes, I confirm the issue too.

**Comment by MegaBaconSticks** [comment]
Just wanted to change the Ring's Voice Language, can confirm, it does indeed hang. On the latest Yuzu Early Access (3782).

**Comment by olzarus** [comment]
Hello,
yes, also can confirm this issue
Yuzu 1533

**Comment by LiukDiihMieu** [comment]
Confirmed using mainline 1541. Is there any way to circumvent this?

**Comment by Alkaid-Benetnash** [comment]
Hi all,
I had the same issue. According to the log, it seems to be related to the "unimplemented websession" feature in function `WebBrowser::ExecuteInteractive`. This particular feature was intentionally left unimplemented since the very beginning of current version of Web Browser related applet (#5042). There is also a Super Mario 3D All-Stars specific websession workaround (#8135). But I don't know whether there could be a similar workaround for Ring Fit or we really need a more serious websession implementation (could we port from the [libnx websession](https://github.com/switchbrew/libnx/blob/99edb1c0f0feceb9890d113ce3f3760740b6ea4f/nx/source/applets/web.c#L967))?

PS: I am a user, not a dev. It would be great if the Web Browser Applet's original dev @Morph1984 could comment on the missing "WebSession"

**Comment by DragoonAethis** [comment]
There are 3 issues here:

- The game opens the web applet with LibraryAppletMode::AllForegroundInitiallyHidden
- Actually requires `window.nx.SendMessage` and friends to work
- It looks like the UI isn't able to open even when forced into the foreground by modifying the emulator - if you take a look into the QtWebEngine debugger (run the emulator with `QTWEBENGINE_REMOTE_DEBUGGING=9000 yuzu`) the loaded `romfs/html-document/ToolMenu.htdocs/index.html` file references .css/.js files with a path like `/static/js/main.ToolMenu.js`. QtWebEngine can't open that path, and it's relative to the .htdocs root. If you remove the first slash, the UI opens in a normal web browser, with a [partially broken UI in Korean (...?) ](https://github.com/yuzu-emu/yuzu/assets/1546665/b2ee1161-6c15-4223-8db6-21366acefcef). (It looks like the browser ignores modded RomFS contents for web content... or maybe something else is broken as well, dunno.)

**Comment by Alkaid-Benetnash** [comment]
Tested PR #12756. It is not merged yet so no public builds available, but I can confirm that PR solves this issue. The "My Menu" page can be accessed within the game.
